provisioning/travis_ci_and_cd/build_and_push_to_ecr.sh

```diff
@@ -0,0 +1,40 @@
+#! /bin/bash
```

Comment:
This script is ALWAYS run after passing tests but you only go inside the `if` conditions if:

* The current branch == "development" || "production" || a branch with the same name as the env var "HOT_DEPLOY_BRANCH"  

That allows us to deploy test/feature branches to our development environment without pushing them to our `development` branch.

This script builds the docker image and pushes it to ECR.
An upcoming PR will do the actual deployment.

Some of the variables here are provided by Travis (see their [convenience variables documentation](https://docs.travis-ci.com/user/environment-variables/#Convenience-Variables)), others are defined in our [Travis settings for this project](https://travis-ci.org/NYPL/fedora_ingest_rails/settings).